### PR TITLE
Pass additional fields to agent create

### PR DESCRIPTION
**Pull request by noahjax**
## Tracking issue
_https://github.com/flyteorg/flyte/issues/<number>_

## Why are the changes needed?
With the update to flytekit 1.11.0 (specifically this [PR](https://github.com/flyteorg/flytekit/pull/2146/files#diff-9f7af27264f8773b069e8200804c224fe19a6fcaaf9dc33edc644f5351cbb3beL103)) and creation of `AsyncAgentBase`, `output_prefix` was removed from the `create` function for agents. This is problematic for our use case, and in general seems like it is moving things towards less flexible agents.

We would also like to be able to consume the task execution metadata in our create function if possible, and I may tack that on to this PR later.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: flytekit/extend/backend/base_agent.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def metadata_type(self) -> ResourceMeta:
         return self._metadata_type
 
     @abstractmethod
-    def create(self, task_template: TaskTemplate, inputs: Optional[LiteralMap], **kwargs) -> ResourceMeta:
+    def create(self, task_template: TaskTemplate, output_prefix: str, inputs: Optional[LiteralMap], **kwargs) -> ResourceMeta:
```
Comment (pingsutw) [inline flytekit/extend/backend/base_agent.py:148]: Could we add the output_prefix only to the base class? since other agents in the flytekit don't use it

**Review by pingsutw — APPROVED** [review]

**Comment by noahjax** [comment]
@pingsutw At some point I'd like to also add TaskExecutionMetadata to agent create (more context [here](https://flyte-org.slack.com/archives/CP2HDHKE1/p1710537982845049?thread_ts=1710428609.845329&cid=CP2HDHKE1)). Should I add that to this PR or address it in a followup?

**Comment by pingsutw** [comment]
@noahjax yes, you can add that to this PR.

**Comment by pingsutw** [comment]
Sure, either works from me

**Comment by noahjax** [comment]
[This](https://github.com/flyteorg/flytekit/pull/2282) is what I initially tried for adding `TaskExecutionMetadata`. 

I don't see a way to add it without making a breaking change to `AsyncAgentBase.create`. Is this ok if I update all of its implementations?

**Comment by noahjax** [comment]
@pingsutw Everything look ok to you?

**Review by pingsutw — APPROVED** [review]

**Comment by pingsutw** [comment]
LGTM, thank you.

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]
LGTM, thank you @ddl-ebrown @noahjax

**Pull request merged**